packages/synapse-react-client/src/components/GenericCard/PortalDOI/PortalDOI.tsx

```diff
@@ -0,0 +1,75 @@
+import CopyToClipboardIcon from '@/components/CopyToClipboardIcon'
+import { CreateOrUpdateDoiModal } from '@/components/doi/CreateOrUpdateDoiModal'
+import { useGetDOIAssociation } from '@/synapse-queries/doi/useDOI'
+import { useGetUserPortalPermissions } from '@/synapse-queries/portal/usePortal'
+import { CreateTwoTone } from '@mui/icons-material'
+import { IconButton, Skeleton, Tooltip, Typography } from '@mui/material'
+import Link from '@mui/material/Link'
+import { useState } from 'react'
+
+export type PortalDOIProps = {
+  /** The ID of the portal created with https://rest-docs.synapse.org/rest/POST/portal.html */
+  portalId: string
+  /** The ID of the resource. For Portals, this is a string that serializes the resourceType and all key information for
+   *  the resource */
+  resourceId: string
+}
+
+/**
+ * Component used to view/create/edit a DOI for a portal resource.
+ */
+function PortalDOI(props: PortalDOIProps) {
+  const [showEditModal, setShowEditModal] = useState(false)
+  const { portalId, resourceId } = props
+
+  const { data: canMintDoi } = useGetUserPortalPermissions(portalId, {
```

Comment:
Since this is shown for every card on the page, would it make sense to use react-intersection-observer (to only load when it is close to the viewport)?

Reply:
Great idea!

Reply:
I think I will add the check to the logic that adds the label rather than this one